### PR TITLE
Abandon adopting orphaned jobs

### PR DESCRIPTION
**Pull request by iszulcdeepsense**
Closes https://github.com/TheRacetrack/racetrack/issues/163
Closes https://github.com/TheRacetrack/racetrack/issues/134

Summary of changes:
- Racetrack doesn't adopt "orphaned" jobs any longer. Lifecycle-supervisor just prints the warning log about the orphaned job.
- a new Prometheus metric is exposed - the number of jobs by status.
- This metric is the first metric of lifecycle-supervisor, so it was added to prometheus config.
- In Grafana dashboards, there's a new panel monitoring the number of jobs with the particular status.
  
Screenshot from 2023-03-07 13-24-30

### CODE REVIEW

File: lifecycle/lifecycle/job/registry.py [hunk 1]
```diff
@@ -124,7 +126,7 @@ def sync_registry_jobs(config: Config, plugin_engine: PluginEngine):
         cluster_jobs = _generate_job_map(list_cluster_jobs(config, plugin_engine))
         registry_jobs = _generate_job_map(list_job_registry(config))
         logger.debug(f'Found {len(cluster_jobs)} jobs in the cluster, {len(registry_jobs)} in the database')
-        job_status_count: Dict[str, int] = defaultdict(int)
+        job_status_count: dict[str, int] = defaultdict(int)
```
Comment (mpolomdeepsense) [inline lifecycle/lifecycle/job/registry.py:129]: Why this change? It doesn't change anything and you have to write an additional import for it to work, and since you are still importing `Dict` from `typing` (used at the bottom of this file) this change seems even more meaningless.

Comment (iszulcdeepsense) [reply lifecycle/lifecycle/job/registry.py:129]: Python 3.9 introduced this https://peps.python.org/pep-0585/. Starting from Python 3.9, using the generic types is now preferred rather than their `typing` counterpart. 
So I keep replacing it once I see it somewhere around.

**Review by Koshmaar — APPROVED** [review]
LGTM. Just one thing that could be improved, is adding some developer doc about the design change "RT doesn't adopt orphaned jobs, because (...), we do it differently (...)". Not sure if you're documenting this kind of logic, but I imagine that if I was a new developer starting in this project, getting to know such details of logic would involve long grepping through code.

File: lifecycle/lifecycle/job/registry.py (hunk 1 above)
Comment (iszulcdeepsense) [reply lifecycle/lifecycle/job/registry.py:129]: As you said, there is still `typing.Dict` import, so perhaps I should replace all occurrences in the whole file to get rid of it.

**Comment by Koshmaar** [comment]
Yeah, agree that making such documentation is energy costly, and we should be conserving energy. Also docs get outdated faster than comments, so what you propose here is ok for me. But if I may clarify something, I didn't mean putting details of all functions in the docs, just high level overview, ie. https://rep.erst.dk/git/kubernilla/recordkeeper/-/tree/master/graph_builder#implementation-details
